hpx/lcos/local/sliding_semaphore.hpp

```diff
@@ -62,6 +62,21 @@ namespace hpx { namespace lcos { namespace local
           : sem_(max_difference, lower_limit)
         {}
 
+        /// \brief Set/Change the difference that will cause the semaphore to trigger
+        ///
+        /// \param max_difference
+        ///                 [in] The max difference between the upper limit
+        ///                 (as set by wait()) and the lower limit (as set by
+        ///                 signal()) which is allowed without suspending any
+        ///                 thread calling wait().
+        /// \param lower_limit  [in] The initial lower limit.
+        void set_max_difference(std::int64_t max_difference,
+                std::int64_t lower_limit = 0)
+        {
+            std::unique_lock<mutex_type> l(mtx_);
+            sem_.set_max_difference(l, max_difference, lower_limit);
```

Comment:
This will require to move the lock. Alternatively you can turn the first argument of `set_max_difference` into a reference.

Reply:
sorry. bad cut'n'paste. there are several mistakes in there actually (in the other functions too). good catch.